include/ensmallen_bits/sgd/update_policies/weight_decay_momentum_update.hpp

```diff
@@ -15,15 +15,18 @@
 namespace ens {
 
 /**
- * Momentum update for Stochastic Gradient Descent with weight decay.
+ * De-Coupled Weight Decay Policy for SGD with Momentum (SGDR).
```

Comment:
Couple to nitpicks:

1. Doesn't SGDR suggest SGD with restarts? If this is not a mistake, ignore me.
2. Does it need to be named de-coupled weight decay policy? The way it's implemented it's just a momentum update which also has weight decay. This could be a potential source of confusion later. Just a thought.

Reply:
1. Yeah SGDR is without decay 
2. Weight Decay is kind of confusing for SGD as both L2 and Weight Decay are equivalent as there is only 1 training value per step , the change will make sense for batches . It doesnt matter if you implment this in Gradient Calculation or during Update but maybe if this paper is built upon in the future then I think this format will atleast provide more backbone as the paper states that this method of Updating can be used for all optimisers , they tried it for SGD and Adam.  
I hope you understand what I am trying to say.